src/circuit/note_commit.rs

```diff
@@ -2230,129 +2485,196 @@ mod tests {
                     Value::known(rcm),
                 )?;
 
+                let asset = NonIdentityPoint::new(
+                    ecc_chip.clone(),
+                    layouter.namespace(|| "witness asset"),
+                    self.asset.map(|asset| asset.cv_base().to_affine()),
+                )?;
+
+                let is_native_asset = assign_free_advice(
```

Comment:
This part is repeated in `circuit.rs`, consider encapsulation.

Reply:
Created assign_is_native_asset